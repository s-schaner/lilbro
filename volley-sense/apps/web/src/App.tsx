import { EventBusProvider } from '@context/EventBusContext';
import { ModuleProvider } from '@context/ModuleContext';
import Dashboard from '@pages/Dashboard';

const App = () => {
  return (
    <ModuleProvider>
      <EventBusProvider>
        <Dashboard />
      </EventBusProvider>
    </ModuleProvider>
  );
};

export default App;
