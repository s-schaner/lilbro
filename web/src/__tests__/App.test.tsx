import { render, screen } from '@testing-library/react';
import { afterEach, beforeEach, expect, test, vi } from 'vitest';
import App from '../App';

const eventsPayload = {
  players: [
    { id: 'p1', name: 'Test Player', position: 'Setter' },
    { id: 'p2', name: 'Another Player', position: 'Libero' },
  ],
  events: [
    { id: 'e1', label: 'Serve', outcome: 'ace', timestamp: 10, player_id: 'p1', video_time: '' },
  ],
};

const formationPayload = { formation: { front_row: ['p1'], back_row: ['p2'] } };
const modulesPayload = {
  modules: [
    { name: 'analysis', status: 'healthy', enabled: true },
    { name: 'ingest', status: 'healthy', enabled: true },
  ],
};

type FetchArgs = [input: RequestInfo | URL, init?: RequestInit];

beforeEach(() => {
  vi.spyOn(globalThis, 'fetch').mockImplementation(async (...args: FetchArgs) => {
    const [input] = args;
    const url = typeof input === 'string' ? input : input.url;
    if (url.endsWith('/events')) {
      return new Response(JSON.stringify(eventsPayload), { status: 200 });
    }
    if (url.endsWith('/events/formation')) {
      return new Response(JSON.stringify(formationPayload), { status: 200 });
    }
    if (url.endsWith('/modules/health')) {
      return new Response(JSON.stringify(modulesPayload), { status: 200 });
    }
    if (url.endsWith('/ingest/health')) {
      return new Response(JSON.stringify({ ok: true, module: 'ingest' }), { status: 200 });
    }
    if (url.includes('/ingest/status')) {
      return new Response(
        JSON.stringify({ status: 'ready', stage: 'ready', progress: 100 }),
        { status: 200 },
      );
    }
    return new Response('{}', { status: 200 });
  });
});

afterEach(() => {
  vi.restoreAllMocks();
});

test('renders VolleySense layout snapshot', async () => {
  const { asFragment } = render(<App />);
  await screen.findByText('VolleySense Console');
  await screen.findByText('Ingest: ready');
  expect(asFragment()).toMatchSnapshot();
});
