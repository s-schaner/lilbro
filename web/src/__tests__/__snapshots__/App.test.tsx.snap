// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders VolleySense layout snapshot 1`] = `
<DocumentFragment>
  <div
    class="min-h-screen bg-slate-950 text-slate-100"
  >
    <header
      class="border-b border-slate-800 bg-slate-950/60 backdrop-blur"
    >
      <div
        class="mx-auto flex max-w-7xl items-center justify-between px-6 py-4"
      >
        <div
          class="flex items-center gap-3 text-lg font-semibold"
        >
          <svg
            class="lucide lucide-video h-6 w-6 text-brand"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m22 8-6 4 6 4V8Z"
            />
            <rect
              height="12"
              rx="2"
              ry="2"
              width="14"
              x="2"
              y="6"
            />
          </svg>
           VolleySense Console
        </div>
        <div
          class="flex items-center gap-3 text-xs text-slate-400"
        >
          <svg
            class="lucide lucide-layers h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m12.83 2.18a2 2 0 0 0-1.66 0L2.6 6.08a1 1 0 0 0 0 1.83l8.58 3.91a2 2 0 0 0 1.66 0l8.58-3.9a1 1 0 0 0 0-1.83Z"
            />
            <path
              d="m22 17.65-9.17 4.16a2 2 0 0 1-1.66 0L2 17.65"
            />
            <path
              d="m22 12.65-9.17 4.16a2 2 0 0 1-1.66 0L2 12.65"
            />
          </svg>
           Modules online: 1
          <button
            class="rounded-full border px-3 py-1 text-xs font-medium transition focus:outline-none focus:ring-2 focus:ring-brand border-emerald-500/40 bg-emerald-500/10 text-emerald-200"
            type="button"
          >
            Ingest: ready
          </button>
          <button
            class="rounded-md bg-brand px-3 py-2 text-sm font-semibold text-white"
          >
            <svg
              class="lucide lucide-upload-cloud mr-2 inline h-4 w-4"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4 14.899A7 7 0 1 1 15.71 8h1.79a4.5 4.5 0 0 1 2.5 8.242"
              />
              <path
                d="M12 12v9"
              />
              <path
                d="m16 16-4-4-4 4"
              />
            </svg>
             Upload
          </button>
        </div>
      </div>
    </header>
    <main
      class="mx-auto flex w-full max-w-7xl flex-1 gap-6 px-6 py-6"
    >
      <div
        class="flex w-72 flex-col gap-4"
      >
        <div
          class="module-card space-y-4"
        >
          <div
            class="flex items-center justify-between"
          >
            <h2
              class="flex items-center gap-2 text-sm font-semibold uppercase tracking-wide text-slate-300"
            >
              <svg
                class="lucide lucide-wand2 h-4 w-4"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m21.64 3.64-1.28-1.28a1.21 1.21 0 0 0-1.72 0L2.36 18.64a1.21 1.21 0 0 0 0 1.72l1.28 1.28a1.2 1.2 0 0 0 1.72 0L21.64 5.36a1.2 1.2 0 0 0 0-1.72Z"
                />
                <path
                  d="m14 7 3 3"
                />
                <path
                  d="M5 6v4"
                />
                <path
                  d="M19 14v4"
                />
                <path
                  d="M10 2v2"
                />
                <path
                  d="M7 8H3"
                />
                <path
                  d="M21 16h-4"
                />
                <path
                  d="M11 3H9"
                />
              </svg>
               Event Trainer
            </h2>
            <button
              class="rounded-md bg-brand px-3 py-1 text-xs font-semibold text-white disabled:opacity-50"
            >
              Generate
            </button>
          </div>
          <div
            class="space-y-3 text-sm"
          >
            <label
              class="block text-xs uppercase text-slate-400"
              for="label"
            >
              Label
            </label>
            <input
              class="w-full rounded-md border border-slate-700 bg-slate-950 p-2 text-sm"
              id="label"
              value="Quick tempo read"
            />
            <label
              class="block text-xs uppercase text-slate-400"
              for="timestamp"
            >
              Timestamp (s)
            </label>
            <input
              class="w-full rounded-md border border-slate-700 bg-slate-950 p-2 text-sm"
              id="timestamp"
              type="number"
              value="42.5"
            />
          </div>
        </div>
        <div
          class="module-card"
        >
          <h3
            class="text-sm font-semibold uppercase tracking-wide text-slate-300"
          >
            Module Health
          </h3>
          <ul
            class="space-y-2"
          >
            <li
              class="flex items-center justify-between rounded-md bg-slate-900 border border-slate-800 px-3 py-2"
            >
              <span
                class="text-sm font-medium capitalize"
              >
                analysis
              </span>
              <span
                class="flex items-center gap-1 text-xs"
              >
                <svg
                  class="lucide lucide-check-circle h-4 w-4 text-emerald-400"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M22 11.08V12a10 10 0 1 1-5.93-9.14"
                  />
                  <path
                    d="m9 11 3 3L22 4"
                  />
                </svg>
                healthy
              </span>
            </li>
          </ul>
        </div>
      </div>
      <div
        class="flex flex-1 flex-col gap-4"
      >
        <section
          class="relative h-72"
        >
          <div
            class="relative h-full w-full overflow-hidden rounded-xl border border-slate-800 bg-slate-950"
          >
            <video
              class="h-full w-full object-contain"
              controls=""
              playsinline=""
            >
              Your browser does not support the video tag.
            </video>
            <canvas
              aria-hidden="true"
              class="pointer-events-none absolute inset-0 h-full w-full"
            />
            <div
              class="absolute inset-0 flex flex-col items-center justify-center gap-3 text-center"
            >
              <div
                class="text-sm text-slate-400"
              >
                Upload a clip to begin video analysis.
              </div>
            </div>
            <canvas
              aria-hidden="true"
              class="hidden"
            />
          </div>
        </section>
        <section
          class="module-card"
        >
          <div
            class="relative h-20"
          >
            <div
              class="absolute left-0 right-0 top-1/2 h-0.5 -translate-y-1/2 bg-slate-700"
            />
            <div
              class="absolute top-2 flex -translate-x-1/2 flex-col items-center"
              style="left: 16.666666666666664%;"
            >
              <div
                class="h-6 w-6 rounded-full border border-brand bg-brand/30 text-center text-xs leading-6"
              >
                â€¢
              </div>
              <span
                class="mt-2 text-xs text-slate-400"
              >
                Serve
              </span>
            </div>
          </div>
        </section>
        <div
          class="module-card space-y-3"
        >
          <h3
            class="flex items-center gap-2 text-sm font-semibold uppercase tracking-wide text-slate-300"
          >
            <svg
              class="lucide lucide-camera h-4 w-4"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.5 4h-5L7 7H4a2 2 0 0 0-2 2v9a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2V9a2 2 0 0 0-2-2h-3l-2.5-3z"
              />
              <circle
                cx="12"
                cy="13"
                r="3"
              />
            </svg>
             ScreenSnap
          </h3>
          <p
            class="text-xs text-slate-400"
          >
            Capture a mock frame and send to the API with a contextual system prompt for structured feedback.
          </p>
          <div
            class="flex gap-2"
          >
            <button
              class="rounded-md bg-slate-800 px-3 py-2 text-sm hover:bg-slate-700"
            >
              Capture Frame
            </button>
            <button
              class="rounded-md bg-brand px-3 py-2 text-sm font-semibold text-white disabled:opacity-50"
              disabled=""
            >
              Send to ScreenSnap
            </button>
          </div>
        </div>
      </div>
      <div
        class="w-80 space-y-4"
      >
        <div
          class="module-card"
        >
          <div
            class="flex gap-2"
          >
            <button
              class="tab-button tab-button-active"
            >
              Players
            </button>
            <button
              class="tab-button bg-slate-800 text-slate-300"
            >
              Events
            </button>
            <button
              class="tab-button bg-slate-800 text-slate-300"
            >
              Formation
            </button>
            <button
              class="tab-button bg-slate-800 text-slate-300"
            >
              Insights
            </button>
          </div>
          <div
            class="mt-4"
          >
            <div
              class="module-card space-y-3"
            >
              <h3
                class="flex items-center gap-2 text-sm font-semibold uppercase tracking-wide text-slate-300"
              >
                <svg
                  class="lucide lucide-circle-user-round h-4 w-4"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M18 20a6 6 0 0 0-12 0"
                  />
                  <circle
                    cx="12"
                    cy="10"
                    r="4"
                  />
                  <circle
                    cx="12"
                    cy="12"
                    r="10"
                  />
                </svg>
                 Roster
              </h3>
              <ul
                class="space-y-2 text-sm"
              >
                <li
                  class="flex items-center justify-between rounded-md bg-slate-800 px-3 py-2"
                >
                  <span>
                    Test Player
                  </span>
                  <span
                    class="text-xs text-slate-400"
                  >
                    Setter
                  </span>
                </li>
                <li
                  class="flex items-center justify-between rounded-md bg-slate-800 px-3 py-2"
                >
                  <span>
                    Another Player
                  </span>
                  <span
                    class="text-xs text-slate-400"
                  >
                    Libero
                  </span>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </main>
    <footer
      class="border-t border-slate-800 bg-slate-950/50"
    >
      <div
        class="mx-auto flex max-w-7xl items-center justify-between px-6 py-4"
      >
        <div
          class="text-xs uppercase tracking-wide text-slate-500"
        >
          Exports
        </div>
        <div
          class="flex gap-3"
        >
          <button
            class="rounded-md border border-slate-700 bg-slate-900 px-3 py-2 text-xs uppercase tracking-wide hover:border-brand"
          >
            report
          </button>
          <button
            class="rounded-md border border-slate-700 bg-slate-900 px-3 py-2 text-xs uppercase tracking-wide hover:border-brand"
          >
            highlights
          </button>
          <button
            class="rounded-md border border-slate-700 bg-slate-900 px-3 py-2 text-xs uppercase tracking-wide hover:border-brand"
          >
            coach-notes
          </button>
        </div>
        <div
          class="text-xs text-slate-400"
        >
          Select an export to queue a download link.
        </div>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
